ZeldaFullEditor/ZSCustomOverworld.asm

```diff
@@ -3134,19 +3163,9 @@ ReplaceBGColor:
 
     REP #$20 ; Set A in 16bit mode.
 
-    ; TODO: Investigate: Saving the Y may not be necessary, done just in case.
-    PHY
-
-    ; Get area code and times it by 2.
+    ; Get area code and times it by 2. Get the color.
     LDA.b $8A : ASL : TAX
-
-    ; Get the color.
-    LDA.w Pool_BGColorTable, X
-
-    ; Set the BG color.
-    STA.l $7EC500 : STA.l $7EC540
-
-    TAY ; Save the color.
+    LDA.w Pool_BGColorTable, X : PHA
```

Comment:
Why was this function changed? I'd just like to understand it more since it influences the day/night color replacement stuff. I also havent tried the updated version yet.

Reply:
dang i should have answered this question 3 weeks ago lol
iirc I just found another case where this function was used and it needed to be handled differently. Sometimes you need to set the buffer color, sometimes the actual color, and other times both.